include/uxr/agent/utils/CLI.hpp

```diff
@@ -0,0 +1,484 @@
+                attr.c_iflag &= unsigned(~INLCR);   // No map NL to CR.
+                attr.c_iflag &= unsigned(~ICRNL);   // No map CR to NL.
+
+                /* Setting OUTPUT OPTIONS. */
+                attr.c_oflag &= unsigned(~OPOST);   // Set raw output.
+
+                /* Setting OUTPUT CHARACTERS. */
+                attr.c_cc[VMIN] = 10;
+                attr.c_cc[VTIME] = 1;
+
+                /* Get baudrate. */
+                speed_t baudrate = getBaudRate(baudrate_opt_.get_baudrate().c_str());
+
+                /* Setting BAUD RATE. */
+                cfsetispeed(&attr, baudrate);
+                cfsetospeed(&attr, baudrate);
+
+                if (0 == tcsetattr(fd, TCSANOW, &attr))
+                {
+                    server_.reset(new eprosima::uxr::SerialServer(fd, 0, common_opts_.middleware_opt_.get_kind()));
+                    rv = server_->run();
+                }
+            }
+        }
+        return rv;
+    }
+
+private:
+    std::string dev_;
+    CLI::Option* cli_opt_;
+    BaudrateOpt baudrate_opt_;
+    CommonOpts common_opts_;
+};
+
+/*************************************************************************************************
+ * Pseudo-Serial CLI Subcommand
+ *************************************************************************************************/
+class PseudoSerialSubcommand : public Subcommand
+{
+public:
+    PseudoSerialSubcommand(CLI::App& app)
+        : Subcommand{app, "pseudo-serial", "Launch a Pseudo-Serial server", common_opts_}
+        , baudrate_opt_{*cli_subcommand_}
+        , common_opts_{*cli_subcommand_}
+    {
+        cli_subcommand_->callback(std::bind(&PseudoSerialSubcommand::launch_server, this));
+    }
+
+private:
+    bool launch_server() final
+    {
+        bool rv = false;
+
+        /* Open pseudo-terminal. */
+        char* dev = nullptr;
+        int fd = posix_openpt(O_RDWR | O_NOCTTY);
+        if (-1 != fd)
+        {
+            if (grantpt(fd) == 0 && unlockpt(fd) == 0 && (dev = ptsname(fd)))
+            {
+                struct termios attr;
+                tcgetattr(fd, &attr);
+                cfmakeraw(&attr);
+                tcflush(fd, TCIOFLUSH);
+                tcsetattr(fd, TCSANOW, &attr);
+
+                /* Get baudrate. */
+                speed_t baudrate = getBaudRate(baudrate_opt_.get_baudrate().c_str());
+
+                /* Setting BAUD RATE. */
+                cfsetispeed(&attr, baudrate);
+                cfsetospeed(&attr, baudrate);
+
+                /* Log. */
+                std::cout << "Pseudo-Serial device opend at " << dev << std::endl;
+
+                /* Run server. */
+                server_.reset(new eprosima::uxr::SerialServer(fd, 0x00, common_opts_.middleware_opt_.get_kind()));
+                rv = server_->run();
+            }
+        }
+
+        return rv;
+    }
+
+private:
+    BaudrateOpt baudrate_opt_;
+    CommonOpts common_opts_;
```

Comment:
Could this be in the base class, `Subcommand`?.

Reply:
It requires a refactor, I add a TODO.